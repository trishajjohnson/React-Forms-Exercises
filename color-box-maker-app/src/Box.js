import React from "react";

function Box({ id, handleRemove, backgroundColor="red", width=5, height=5 }) {
    
    return (
        <div>
            <div
                style={{
                    backgroundColor: backgroundColor,
                    width: `${width}px`,
                    height: `${height}px`
                }} 
            />
            <button onClick={() => handleRemove(id)} data-testid="add-button">X</button>
        </div> 
    );
}
  
  export default Box;
  