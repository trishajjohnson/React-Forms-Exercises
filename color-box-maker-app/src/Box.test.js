import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

// Smoke test for Box component
it("should render without crashing", function() {
  render(<Box />)
});

// Snapshot test for Box component
it("should match snapshot", function() {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});