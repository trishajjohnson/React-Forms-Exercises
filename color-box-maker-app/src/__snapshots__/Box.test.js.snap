// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="background-color: red; width: 5px; height: 5px;"
    />
    <button
      data-testid="add-button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
